packages/documentation/src/pages/getting-started/common-tasks/new-end-to-end-test.mdx

```diff
@@ -0,0 +1,153 @@
+---
+title: Write an end-to-end test
+tags: Nightwatch, mockapi, e2e
+---
+
+# Write an end-to-end test
+
+Front end engineers use end-to-end (e2e) tests in `vets-website` to validate multipage applications with client-side routing. They are primarily used to assert that:
+- client applications render their inputs
+- client-side navigation occurs when the required fields are populated
+
+## end-to-end testing overview
+
+- `vets-website` uses [Nightwatch](https://nightwatchjs.org) to run the tests and provide the browser client
+- end-to-end tests are **collocated in application folder** with the application they test
+- Two node apps run with the end-to-end tests:
+  - `mockapi.js` - hosts mock responses (see [Mocking API responses](#mocking-api-responses))
+  - `test-server.js` - builds and hosts `vets-website` with support for client-side routes
```

Comment:
I don't think I would say this "builds" vets-website, but it does run a local webserver that handles client-side routes, like you say. You have to build the site yourself if you're not running `yarn watch`, otherwise the e2e tests won't work.